pkg/apis/sources/v1alpha2/apiserver_conversion_test.go

```diff
@@ -32,3 +40,294 @@ func TestApiServerConversionBadType(t *testing.T) {
+						MatchLabels: map[string]string{"A1": "K1"},
+					},
+				}, {
+					APIVersion: "A2",
+					Kind:       "K2",
+				}},
+				EventMode: "Ref",
+			},
+			Status: ApiServerSourceStatus{
+				SourceStatus: duckv1.SourceStatus{
+					Status: duckv1.Status{
+						ObservedGeneration: 1,
+						Conditions: duckv1.Conditions{{
+							Type:   "Ready",
+							Status: "Unknown",
+						}},
+					},
+					SinkURI: sinkUri,
+				},
+			},
+		},
+	}, {name: "full",
+		in: &ApiServerSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:       "apiserver-name",
+				Namespace:  "apiserver-ns",
+				Generation: 17,
+			},
+			Spec: ApiServerSourceSpec{
+				SourceSpec: duckv1.SourceSpec{
+					Sink: sink,
+					CloudEventOverrides: &duckv1.CloudEventOverrides{
+						Extensions: map[string]string{
+							"foo": "bar",
+							"baz": "baf",
+						},
+					},
+				},
+				Resources: []APIVersionKindSelector{{
+					APIVersion: "A1",
+					Kind:       "K1",
+					LabelSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"foo": "bar",
+						},
+						MatchExpressions: []metav1.LabelSelectorRequirement{{
+							Key:      "aKey",
+							Operator: metav1.LabelSelectorOpIn,
+							Values:   []string{"the", "house"},
+						}},
+					},
+				}, {
+					APIVersion: "A2",
+					Kind:       "K2",
+				}},
+				ResourceOwner: &APIVersionKind{
+					APIVersion: "custom/v1",
+					Kind:       "Parent",
+				},
+				EventMode:          "Resource",
+				ServiceAccountName: "adult",
+			},
+			Status: ApiServerSourceStatus{
+				SourceStatus: duckv1.SourceStatus{
+					Status: duckv1.Status{
+						ObservedGeneration: 1,
+						Conditions: duckv1.Conditions{{
+							Type:   "Ready",
+							Status: "True",
+						}},
+					},
+					SinkURI: sinkUri,
+				},
+			},
+		},
+	}}
+	for _, test := range tests {
+		for _, version := range versions {
+			t.Run(test.name, func(t *testing.T) {
+				ver := version
+				if err := test.in.ConvertTo(context.Background(), ver); err != nil {
+					t.Errorf("ConvertTo() = %v", err)
+				}
+
+				got := &ApiServerSource{}
+
+				if err := got.ConvertFrom(context.Background(), ver); err != nil {
+					t.Errorf("ConvertFrom() = %v", err)
+				}
+				if diff := cmp.Diff(test.in, got); diff != "" {
+					t.Errorf("roundtrip (-want, +got) = %v", diff)
+				}
+			})
+		}
+	}
+}
+
+// This tests round tripping from a higher version -> v1alpha1 and back to the higher version.
+func TestApiServerSourceConversionRoundTripDown(t *testing.T) {
+	path, _ := apis.ParseURL("/path")
+	sink := duckv1.Destination{
+		Ref: &duckv1.KReference{
+			Kind:       "Foo",
+			Namespace:  "Bar",
+			Name:       "Baz",
+			APIVersion: "Baf",
+		},
+		URI: path,
+	}
+	sinkURI, _ := apis.ParseURL("http://example.com/path")
+
+	ceOverrides := duckv1.CloudEventOverrides{
+		Extensions: map[string]string{
+			"foo": "bar",
+			"baz": "baf",
+		},
+	}
+
+	tests := []struct {
+		name string
+		in   apis.Convertible
```

Comment:
Changing this with `*v1beta1.ApiServerSource` could make things easier later.
Examples I saw so far are like that.
But not a big deal.

Reply:
I followed the https://github.com/knative/eventing/blob/6caed6a461ed6d930b1069dc235a4f3e019cabc3/pkg/apis/sources/v1alpha1/apiserver_conversion_test.go#L222 here. I think at first we use `apis.Convertible` so that provide different versions of sources as `in`

Reply:
For now, I will keep apis.Convertible since it's more flexiable